packages/patrol_finders/lib/src/custom_finders/patrol_tester.dart

```diff
@@ -116,17 +117,60 @@ const defaultScrollMaxIteration = 15;
 /// <https://dart.dev/guides/language/language-tour#callable-classes>
 class PatrolTester {
   /// Creates a new [PatrolTester] which wraps [tester].
-  const PatrolTester({
+  PatrolTester({
     required this.tester,
     required this.config,
   });
 
   /// Global configuration of this tester.
   final PatrolTesterConfig config;
 
+  /// Logs a message to the patrol log.
+  final PatrolLogWriter _patrolLog = PatrolLogWriter();
+
   /// Flutter's widget tester that this [PatrolTester] wraps.
   final WidgetTester tester;
 
+  /// Wraps a function with a log entry for the start and end of the function.
+  Future<T> wrapWithPatrolLog<T>({
+    required String action,
+    String? value,
+    Finder? finder,
+    required String color,
+    required Function function,
```

Comment:
If you declare this paramteres as:
```suggestion
    required Future<T> Function() function,
```

Reply:
Then you should be able to call the methods inside wrapper without casting:
`      return function();`